src/Float16Array.mjs

```diff
@@ -547,6 +547,37 @@ export class Float16Array {
     return convertToNumber(float16bitsArray[k]);
   }
 
+  /** @see https://tc39.es/proposal-change-array-by-copy/#sec-%typedarray%.prototype.withAt */
+  withAt(index, value) {
+    assertFloat16Array(this);
+    const float16bitsArray = getFloat16BitsArray(this);
+
+    const length = TypedArrayPrototypeGetLength(float16bitsArray);
+    const relativeIndex = ToIntegerOrInfinity(index);
```

Comment:
The current spec uses `IsIntegralNumber` instead, which is different from the `at` method and may change.
related: https://github.com/tc39/proposal-change-array-by-copy/issues/42